### PR TITLE
Added docs for new timezone support added at #6788

### PR DESCRIPTION
**Pull request by zerkms**
I added documentation for #6788.

As a non-native speaker I presume it might be reworded. And I hope as-is it at least expresses my thoughts clearly enough.

### CODE REVIEW

File: man/systemd.time.xml [hunk 1]
```diff
@@ -123,7 +123,11 @@
 
     <para>When parsing, systemd will accept a similar syntax, but expects no timezone specification, unless it is given
     as the literal string <literal>UTC</literal> (for the UTC timezone) or is specified to be the locally configured
-    timezone. Other timezones than the local and UTC are not supported. The weekday specification is optional, but when
+    timezone. Other timezones than the local and UTC are not supported. Starting from v235 it is possible to specify
```
Comment (zdzichu) [inline man/systemd.time.xml]: We tend to document current state of affairs, so “Starting from v235” should be dropped.
It adds little information, duplicated (we have NEWS file for that) and quickly grows into mess of historical information mixed with useful things.

Comment (zerkms) [reply man/systemd.time.xml]: Agree, changed. Thanks.

File: man/systemd.time.xml [hunk 2]
```diff
@@ -122,8 +122,12 @@
...
 
     <para>When parsing, systemd will accept a similar syntax, but expects no timezone specification, unless it is given
-    as the literal string <literal>UTC</literal> (for the UTC timezone) or is specified to be the locally configured
-    timezone. Other timezones than the local and UTC are not supported. The weekday specification is optional, but when
+    as the literal string <literal>UTC</literal> (for the UTC timezone), or is specified to be the locally configured
+    timezone, or the timezone name in the IANA timezone database format. The complete list of timezones
+    supported on your system can be obtained using the <literal>timedatectl list-timezones</literal>
+    (see <citerefentry><refentrytitle>timedatectl</refentrytitle><manvolnum>1</manvolnum></citerefentry>).
+    Using IANA format is recommended over local timezone names, as less prone to errors (eg: with local timezone it's possible to
+    specify daylight savnig time in winter, while it's incorrect). The weekday specification is optional, but when
```
Comment (boucman) [inline man/systemd.time.xml]: savnig => saving

Comment (zerkms) [reply man/systemd.time.xml]: thanks. Fixed and squashed all changesets into one.

**Comment by poettering** [comment]
lgtm

**Pull request merged**